sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs

```diff
@@ -214,8 +204,66 @@ private static void WriteStartupCodeExecutorClass(IndentedTextWriter textWriter,
             textWriter.WriteLine("}");
         }
 
-        public void Initialize(GeneratorInitializationContext context)
+        private static bool IsAssemblyWithExtensionStartupAttribute(IAssemblySymbol arg)
         {
+            var extensionStartupAttribute = GetStartupAttributeData(arg);
+
+            return extensionStartupAttribute != null;
+        }
+
+        private static AttributeData? GetStartupAttributeData(IAssemblySymbol assemblySymbol)
+        {
+            var extensionStartupAttribute = assemblySymbol.GetAttributes()
+                .FirstOrDefault(a =>
+                    (a.AttributeClass?.Name.Equals(AttributeTypeName,
+                        StringComparison.Ordinal) ?? false) &&
+                    //Call GetFullName only if class name matches.
+                    a.AttributeClass.GetFullName()
```

Comment:
I think the best practice here is to use the `Compilation` object to get the `INamedTypedSymbol` we want to compare against via name, and then we can compare the symbols directly. See the logger message example [here](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs#L39) And you can see how that instance is used for comparison in that class.

Reply:
Good suggestion. Switched to this approach.